src/Raven.Server/Documents/Indexes/Workers/HandleReferences.cs

```diff
@@ -385,22 +401,44 @@ private void UpdateReferences(TransactionOperationContext indexContext, string c
             }
         }
 
+        private static void RenewTransactionIfNeeded(QueryOperationContext queryContext, Index.CanContinueBatchResult batchContinuationResult, HashSet<Slice> indexed, ref IDisposable readTransaction)
```

Comment:
Why do you use `IDisposable` for the readTransaction instead of the actual transaction type

Reply:
I thought that `OpenReadTransaction()` returns `DocumentsTransaction` object but now I see it returns `IDisposable`